### ISSUE TITLE
add payload cost

### ISSUE THREAD

**Issue by zbraniecki** [FrankYFTang/proposal-intl-extend-timezonename#1]
This proposal will require bundling additional data into JS engines. In order to evaluate the trade-off value we need to estimate the cost.

**Comment by FrankYFTang** [comment]
I take a slightly closer look 


location | VVV | Los Angeles |  
longLocation | VVVV | Los Angeles TIme

in deed would need a lot of extra data. (my prototype were based on those data filter out and show no localized text) I would put down more estimate about that and recommend not to include them. But four other formats looks like won't cost that. much. Stay tune of the number.

**Comment by FrankYFTang** [comment]
Here is my finding about the data size. I put how I get the number in the README.md. I also change my proposal

(notice the below calculation only count the size for the "value", not including the size for the "key", you can estimate the extra size for the key, by using the number of items, the size for the key heavily depend on how the data were encoded)
1. the shortGMT and longGMT use one pattern per locale, so to cover ALL the locales in icu data set (not just the one shipped with Chrome) it took 263 items, total 1826 bytes in UTF8 and gzip compressed 392 bytes together. 
2. to support the shortWall, since it fallback to pre-existing pattern if absent, to cover ALL the locales in icu data set (not just the one shipped with Chrome) it took 332 items, 1719 extra UTF-8 bytes, and gzip 311 bytes together.
3. to support the longWall, since it fallback to pre-existing pattern if absent, to cover ALL the locales in icu data set (not just the one shipped with Chrome) it took 10047 items, 278103 extra UTF-8 bytes, and gzip 69526 bytes together.

Chrome already include all the data for the above 3 for the locales we already supported, so it present no extra data size impact for chrome. Just list above to show what happen if a new implementation will take if they previously only support the minimum data for short and long and they now need to include more data for that if they want to support ALL locales. If they want to support the same number of Locales chrome support, the size will be smaller.

(UTF8 bytes mean the size in UTF8, instead of in UTF16, which could be more bytes, or less bytes)

I also agree location and longLocation will take too much data size increase and change my position and would like to against to support them. The two formats basically use the same amount of data (the long one just use an extra pattern per locale ) support either one or both of them will take 45924 items,  664784 bytes in UTF8 and compressed to  196639 bytes which v8/chrome would like NOT to implement them.

as I mentioned in https://github.com/FrankYFTang/proposal-intl-extend-timezonename/issues/2 I also believe one of the format Steven strawman were for toISOString (since it format ISO8601 for machine to read not for human to read) and not here for ECMA402.

**Comment by FrankYFTang** [comment]
@zbraniecki with the information I put into README.md, do you think I can close this issue?

**Comment by zbraniecki** [comment]
Yeah! Thank you for doing that!

Can you add the `longWall` estimate to the README.md as well? You only put `location` and longLocation`'s payload.

I think the `longWall` remains the only debatable item which you want to include but cost - 278kb on disk and 69kb zipped. Ballpark I can see us debate in Stage 2 over :)

One more thought, maybe for @sffc to consider for his guidelines - increasing payload size has consequences not just on the entry barrier for a strawman "new implementer". It also changes the tradeoff for including more locales.
If an engine ships 50 locales, and the Product evaluates adding 20 more locales, and we increase the cost by 10kb per locale, then the Product is evaluating +200kb which may tilt the decision.
In other words, the larger ECMA-402 payload is, the higher the entry barrier is to be added to locales handled by the engines and in result to be represented on the Web.
As ECMA-402 becomes popular solution to the common problem, the odds of client-side solutions that will handle the additional locales, are getting lower, because the tradeoff of a Product Manager for a website is becoming "0kb to handle 50 locales" vs "Xkb to handle additional locales" where X is non-zero.
That as well increases the odds that the website will only support locales supported by ECMA-402 engines.

I think this realization increases our responsibility to balance the value of adding 5th time zone format vs. supporting 5 more locales on the Web.

**Comment by sffc** [comment]
I think we must consider the two data axes separately.

Beyond the top 50-60 locales, there is a very long tail.  The best way to support that tail is to use the "language pack" model, where browsers or implementations download packs of data when you set up your browser or change your language preference.

As a *general* principle, I don't think it is good to compare "5th time zone format vs. supporting 5 more locales on the Web."  If Firefox *in particular* has that constraint, you should bring it up when we talk about implementability for Stage 3.  I just don't think we will get very far by putting constraints based on theoretical future implementations whose requirements we can only guess.

**Comment by zbraniecki** [comment]
>  If Firefox in particular has that constraint, you should bring it up when we talk about implementability for Stage 3.

I don't think this is Firefox in particular issue. I would expect the same question and tradeoff to be faced by any implementer.

>  I just don't think we will get very far by putting constraints based on theoretical future implementations whose requirements we can only guess.

I don't think this is what I'm bringing. I'm claiming that since payload costs, every payload increase in one axis, costs the other and we should be aware of that.

> Beyond the top 50-60 locales, there is a very long tail.

The cut-off is artificial and arbitrary.

> The best way to support that tail is to use the "language pack" model, where browsers or implementations download packs of data when you set up your browser or change your language preference.

And I hope we'll get to such model, but ICU4C doesn't support it well. I hope to get there with ICU4X, but until that is validated, I think that "vertical" (adding more data per locale) vs "horizontal" (adding more locales) is a real tradeoff.

**Comment by FrankYFTang** [comment]
> Can you add the `longWall` estimate to the README.md as well? You only put `location` and longLocation`'s payload.

I did. It was always there. Notice there are two table. I put that in the top one. since I remove the location and longLocation, I also put the number for that in the lower table since I remove these two from the top table.

**Comment by FrankYFTang** [comment]
Close this issue because the issue is to add payload cost info and I did. We can debate which particular one should keep/remove in other ticket.

**Comment by zbraniecki** [comment]
Ah, I see! Those two tables are a bit confusing, since the former one has a cell with information that is missing in the corresponding cell in the bottom table. But the data is in README, so I'm fine :)

Thank you!
